include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -51,6 +51,8 @@ template <>
 struct TypeToTensorType<uint64_t> { static constexpr ONNXTensorElementDataType type = ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT64; };
 template <>
 struct TypeToTensorType<bool> { static constexpr ONNXTensorElementDataType type = ONNX_TENSOR_ELEMENT_DATA_TYPE_BOOL; };
+template <>
+struct TypeToTensorType<std::string> { static constexpr ONNXTensorElementDataType type = ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING; };
```

Comment:
The reason this template specialization is absent is intentional. This template is used to resolve primitive datatypes in CreateTensor<T>() methods. Those methods take contiguous primitive buffers as arguments, which then fed through a C-API to create Tensors on top of the user supplied buffers and avoid copy. It not possible to feed std::strings as a contiguous buffer, and therefore, create a tensor on a top of it.
For this reason, strings must always be copied and there are separate overloads for that. 
We error on the side of caution and force the user to use non-templated CreateTensor() that takes OrtAllocator and a type enum. Then one can fill the tensor with strings.